test/units/utils/test_unsafe_proxy.py

```diff
@@ -85,3 +85,37 @@ def test_AnsibleUnsafeText():
 
 def test_AnsibleUnsafeBytes():
     assert isinstance(AnsibleUnsafeBytes(b'foo'), AnsibleUnsafe)
+
+
+def test_unwrap_var():
+    value = {
+        'text': AnsibleUnsafeText(u'text'),
+        'bytes': AnsibleUnsafeBytes(b'bytes'),
+        'dict': {
+            'text': AnsibleUnsafeText(u'text'),
+            'bytes': AnsibleUnsafeBytes(b'bytes'),
+        },
+        'list': [
+            AnsibleUnsafeText(u'text'),
+            AnsibleUnsafeBytes(b'bytes'),
+        ],
+        'set': set(
+            (
+                AnsibleUnsafeText(u'text'),
+                AnsibleUnsafeBytes(b'bytes'),
+            )
+        ),
+        'tuple': (
+            AnsibleUnsafeText(u'text'),
+            AnsibleUnsafeBytes(b'bytes'),
+        )
+    }
+
```

Comment:
Maybe mix some unwrapped text and byte strings in here as well to make sure we don't get caught out by that falling through to an immutable sequence check in some future iteration)

Also, i like to always include some non-ascii data when testing that arbitrary strings are doing the right thing.